api/main.go

```diff
@@ -159,33 +234,39 @@ func setupConsumer(brokerAddress []string) error {
 func formatIncomingMessage(incomingMessage *models.IncomingMessage) ([]models.DestinationMessage, error) {
 
 	if incomingMessage == nil {
-		return nil, errors.New("Incoming message is empty")
+		return nil, errors.New("\tIncoming message is empty")
 	}
 
-	destinationMessages := make([]models.DestinationMessage, len(incomingMessage.Message.Partitions))
+	destinationMessages := make([]models.DestinationMessage, 0)
```

Comment:
You can predefine the capacity of the destinationMessages slice make([]models.DestinationMessage, 0, len(incomingMessage.Message.Partitions))

Reply:
Improved, thanks.